### PR TITLE
Helper function to easily get gameobject by their network id

### PR DESCRIPTION
**Pull request by Gentatsu**

### CODE REVIEW

File: Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs [hunk 1]
```diff
@@ -667,5 +667,10 @@ public virtual void SceneReady(Scene scene, LoadSceneMode mode)
 
 			}
 		}
-	}
+
+        public static GameObject GetGameObjectByNetworkId(uint id)
+        {
+            return ((NetworkBehavior)Instance.Networker.NetworkObjects[id].AttachedBehavior).gameObject;
```
Comment (phalasz) [inline Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs]: I personally would either check that the id passed in exists in the dictionary or use the `TryGetValue` method.

File: Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs [hunk 2]
```diff
@@ -667,5 +667,10 @@ public virtual void SceneReady(Scene scene, LoadSceneMode mode)
 
 			}
 		}
-	}
+
+        public static GameObject GetGameObjectByNetworkId(uint id)
```
Comment (phalasz) [inline Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs]: Does this need to be static since you are using `Instance` inside it anyway? Might be cleaner to remove the static keyword and use it through the instance.

Comment (phalasz) [inline Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs]: Also please add some documentation comments for the method.

Comment (Gentatsu) [reply Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs]: I found the static method easier & quicker than having to do `NetworkManager.Instance.GetGameObjectByNetworkId` as opposed to `NetworkManager.GetGameObjectByNetworkId`. 

Since it's a singleton, the static method gives the same results with the same consequences in terms of accessing the `NetworkManager`.

**Comment by phalasz** [comment]
I agree.

File: Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs (hunk 1 above)
Comment (phalasz) [inline Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs]: There should also be checks here that the `Networker` is actually been set. The `Instace` is not null. The `AttachedBehavior` is not null etc.

**Comment by Gentatsu** [comment]
> Would argue that following the code convention would be to not have it static. What do we think people?

Having a quick look at some other functions like: `SendFrame` and the `NetworkBehavior`'s `SkipAttachIds`, which are static, there seems to be a mixture of static and non-static functions. 

I can see why this would not be static, as it goes against an OOP grain, and isn't performing a functional command.

File: Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs (hunk 1 above)
Comment (Gentatsu) [reply Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs]: Would it not be better to let it null error, so it is obvious that these are not set? Only one I would say to check would be `AttachedBehavior`.

File: Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs [hunk 3]
```diff
@@ -668,9 +668,15 @@ public virtual void SceneReady(Scene scene, LoadSceneMode mode)
...
-        public static GameObject GetGameObjectByNetworkId(uint id)
+		/// <summary>
+		/// A helper function to retrieve a GameObject by its network id.
+		/// </summary>
+		/// <param name="id">Network id of the gameobject</param>
+        public GameObject GetGameObjectByNetworkId(uint id)
         {
-            return ((NetworkBehavior)Instance.Networker.NetworkObjects[id].AttachedBehavior).gameObject;
+			NetworkObject foundNetworkObject;
+			Networker?.NetworkObjects?.TryGetValue(id, out foundNetworkObject);
```
Comment (phalasz) [inline Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs]: This is unfortunatelly a C# 6 feature that would break for users not having that available for them.

Comment (Gentatsu) [reply Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs]: Ah lol. I didn't think my commits would automatically become part of the PR. I didn't have my main machine, so was going to squash the commit after testing. 

I'll take them out, though.

Comment (phalasz) [reply Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs]: yeah if you push changes to a branch that has a PR then the PR will automatically have the changes 😉

File: Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs [hunk 4]
```diff
@@ -674,9 +674,18 @@ public virtual void SceneReady(Scene scene, LoadSceneMode mode)
...
         {
-			NetworkObject foundNetworkObject;
-			Networker?.NetworkObjects?.TryGetValue(id, out foundNetworkObject);
-            return ((NetworkBehavior)foundNetworkObject?.AttachedBehavior?).gameObject?;
+            if (Networker == null ) //Only check Networker, as NetworkObjects are always initiliased.
+            {
+                Debug.LogWarning("Networker is null. Check if initiliased");
+                return null;
+            }
+            NetworkObject foundNetworkObject = null;
```
Comment (phalasz) [inline Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs]: Minor thing. Could you add an empty line above this please?

File: Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs [hunk 5]
```diff
@@ -674,9 +674,18 @@ public virtual void SceneReady(Scene scene, LoadSceneMode mode)
...
+            if (!Networker.NetworkObjects.TryGetValue(id, out foundNetworkObject) || foundNetworkObject.AttachedBehavior == null)
+            {
+                Debug.LogWarning("No object found by id or object has no attached behavior.");
+                return null;
+            }
+            return ((NetworkBehavior)foundNetworkObject.AttachedBehavior).gameObject;
```
Comment (phalasz) [inline Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs]: Same here. An empty line above this would make the method more readable.

File: Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs [hunk 6]
```diff
@@ -667,5 +667,25 @@ public virtual void SceneReady(Scene scene, LoadSceneMode mode)
...
+
+		/// <summary>
+		/// A helper function to retrieve a GameObject by its network id.
+		/// </summary>
+		/// <param name="id">Network id of the gameobject</param>
+        public GameObject GetGameObjectByNetworkId(uint id)
+        {
+            if (Networker == null ) //Only check Networker, as NetworkObjects are always initiliased.
+            {
+                Debug.LogWarning("Networker is null. Check if initiliased");
```
Comment (phalasz) [inline Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs]: Maybe reword this to something like `Networker is null. NetworkManager has not been initialised.` 🤔

File: Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs (hunk 4 above)
Comment (Gentatsu) [reply Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs]: Above the NetworkObject declaration & return?

Comment (phalasz) [reply Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/NetworkManager.cs]: yes please. Between the `}` and the line with `NetworkObject`
